### PR TITLE
Average pooling

### PR DESCRIPTION
**Pull request by pvskand**
Adding Avg pooling to the existing pooling layers.
Currently only MaxPool is available in DeepChem.

### CODE REVIEW

File: deepchem/models/tensorgraph/layers.py [hunk 1]
```diff
@@ -2511,6 +2511,47 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
...
+    self.ksize = ksize
+    self.strides = strides
+    self.padding = padding
+    super(AvgPool2D, self).__init__(**kwargs)
+    try:
+      parent_shape = self.in_layers[0].shape
+      self._shape = tuple(
+          None if p is None else p // s for p, s in zip(parent_shape, strides))
+    except:
+      pass
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Raise a `ValueError` here explaining that shape construction has failed rather than failing silently.

Comment (peastman) [reply deepchem/models/tensorgraph/layers.py]: This is how all layers work.  Sometimes the shape just can't be determined.  In that case, leaving `_shape` undefined is the correct thing to do, and `shape()` will throw an exception.

Comment (rbharath) [reply deepchem/models/tensorgraph/layers.py]: Ah, makes sense. Been a while since I've written a new layer :)

**Comment by rbharath** [comment]
LGTM

**Pull request merged**